proxy/proxy.go

```diff
@@ -827,6 +846,25 @@ func WithParams(p Params) *Proxy {
 	}
 }
 
+type otelTransport struct {
+	tr     *http.Transport
+	tracer trace.Tracer
+}
```

Comment:
Why do we need this? Somehow we created proxy span without it before.

Reply:
This is an interesting question and I think it's how OTel is different from OT. Please confirm @lucastt

Reply:
Yeah, maybe this is not necessary. Because the goal of this struct is to inject the span into the request, but we already do this in `net/httpclient.go:406` so the only case this struct would be necessary is if we make requests using this Roundtripper directly, without using skipper http client.